lib/scry/course.rb

```diff
@@ -0,0 +1,163 @@
+      agent = Mechanize.new
+      agent.cookie_jar = YAML::load(cookie_crumbs)
+      course_link = Mechanize::Page::Link.new(
+        {
+          "href" => course_url,
+        },
+        agent,
+        nil,
+      )
+      Course.new(agent, course_link)
+    end
+
+    ##
+    # Creates an export file for the course
+    #
+    # Navigates from the course page to the export course page
+    # and creates an export for the course.
+    #
+    # First it all existing exports, then attempts to create a new export.
+    #
+    # It will wait indefinitely for the export to be created.
+    ##
+    def create_export
+      course_page = @agent.click(@course_link)
+      package_links = course_page.links_with(
+        text: /Packages & Utilities Overview Page/,
+      )
+      if package_links.any?
+        utilities_page = click_link(
+          agent: @agent,
+          page: course_page,
+          text: /Packages & Utilities Overview Page/,
+        )
+        exports_page = click_link(
+          agent: @agent,
+          page: utilities_page,
+          text: /Export\/Archive Course/,
+        )
+        export_links = exports_page.links_with(
+          text: /Export Package/,
+        )
+        if export_links.any?
+          _delete_existing_exports(exports_page, nil)
+          export_page = click_link(
+            agent: @agent,
+            page: exports_page,
+            text: /Export Package/,
+          )
+          exports_page = _process_export_form(export_page)
+          exports = exports_page.links_with(
+            text: "View Basic Log",
+          )
+          _wait_for_export(exports, utilities_page, exports_page)
+        end
+      end
+    end
+
+    ##
+    # Opens the log for an export and checks if it was successful.
+    ##
+    def validate_export(exports_page)
+      links = exports_page.links_with(text: "View Basic Log")
+      if links.empty?
+        raise Scry::ExportFailed, "Links empty #{exports_page.uri}"
+      end
+      url = links.last.attributes["onclick"][/'(.*)'/, 1]
+      log_page = @agent.get(url)
+      text = Nokogiri::HTML(log_page.body).css("div#containerdiv").text
+      !text.match(/error/i)
+    end
+
+    ##
+    # Extracts the download URL for an export
+    ##
+    def download_url(page)
+      download_link = page.links_with(text: "Open").last
+      download_link.href
+    end
+
+    ##
+    # Downloads the export into the given directory.
+    ##
+    def download_export(url, dir)
+      @agent.pluggable_parser["application/zip"] = Mechanize::Download
+      filename = File.basename(URI.parse(url).path)
+      @agent.get(url).save(File.join(dir, filename))
+    end
+
+    ##
+    # :nodoc:
+    # Fills out the export form and submits it.
+    ##
+    def _process_export_form(export_page)
+      export_page.form_with(name: "selectCourse") do |export_form|
+        export_form.radiobuttons[1].check
+        export_form.radiobuttons[3].check
```

Comment:
In the interest of avoiding magic numbers, and maybe making the code a bit easier to understand, maybe create two constants for `1` and `3` here, which describe what the radio buttons are that are being checked.